### PR TITLE
Task2: Add path helper

### PR DESCRIPTION
**Pull request by Source34**
1. Implemented GetPathCount via DP to count grid paths
2. Supports only right/down moves in m×n grid
3. Handles invalid inputs
4. Added tests: small grids, edge cases, invalid input

### CODE REVIEW

File: DataStructures/DataStructures/Grids/PathHelper.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /// <param name="n">Количество столбцов (ширина сетки).</param>
+    /// <returns>Количество уникальных маршрутов.</returns>
+    public static int GetPathCount(int m, int n)
+    {
+        if (m <= 0 || n <= 0)
+            return 0;
+
+        var dp = new int[m, n];
+
+        for (int i = 0; i < m; i++) dp[i, 0] = 1;
```
Comment (oderkov) [inline DataStructures/DataStructures/Grids/PathHelper.cs:23]: Будет ли отличаться сложность если сделать все за один проход? вместо обработки в циклах крайних значений

File: DataStructures/DataStructures.Tests/Grids/PathHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        int result = PathHelper.GetPathCount(m, n);
+        Assert.Equal(expected, result);
+    }
+
+    [Theory]
+    [InlineData(0, 0)]
+    [InlineData(0, 5)]
+    [InlineData(3, 0)]
+    [InlineData(-1, 4)]
```
Comment (i-interested) [inline DataStructures/DataStructures.Tests/Grids/PathHelperTests.cs:28]: Для отрицательных значений не все тест кейсы описаны

Comment (Source34) [reply DataStructures/DataStructures.Tests/Grids/PathHelperTests.cs:28]: Да, согласен
Исправлено!

File: DataStructures/DataStructures/Grids/PathHelper.cs (hunk 1 above)
Comment (Source34) [reply DataStructures/DataStructures/Grids/PathHelper.cs:23]: Нет, сложность останется O(m * n), независимо от того, разбиваем ли мы обработку краёв на отдельные циклы или объединяем в один. У нас в любом случае происходит обход всех m * n ячеек

**Review by oderkov — APPROVED** [review]

**Review by i-interested — APPROVED** [review]

**Review by AlexanderKubankov — APPROVED** [review]
